### PR TITLE
Task/add argparse to hif2a example

### PR DESCRIPTION
**Pull request by badisa**

### CODE REVIEW

File: examples/hif2a/fit_to_multiple_rbfes.py [hunk 1]
```diff
@@ -184,7 +193,6 @@ def _save_forcefield(filename, ff_params):
         prod_steps=configuration.num_prod_steps,
     )
 
-    # TODO: use binding_model.predict rather than binding_model.loss
```
Comment (badisa) [inline examples/hif2a/fit_to_multiple_rbfes.py:187]: @maxentile Based on your last changes I think this comment is no longer relevant?

Comment (maxentile) [reply examples/hif2a/fit_to_multiple_rbfes.py:187]: Good catch -- thank you for spotting and correcting that!

File: examples/hif2a/fit_to_multiple_rbfes.py [hunk 2]
```diff
@@ -147,14 +156,14 @@ def _save_forcefield(filename, ff_params):
     # update the forcefield parameters for a few steps, each step informed by a single free energy calculation
 
     # compute and save the sequence of relative_transformation indices
-    num_epochs = int(np.ceil(num_parameter_updates / len(relative_transformations)))
-    np.random.seed(2021)
+    num_epochs = int(np.ceil(args.param_updates / len(relative_transformations)))
+    np.random.seed(args.seed)
```
Comment (maxentile) [inline examples/hif2a/fit_to_multiple_rbfes.py:160]: Nice! Good idea to make the random seed configurable as well (e.g. when we wish to run replicates that differ only in the random seed)

**Review by maxentile — APPROVED** [review]
Looks great!

Makes several important parameters configurable rather than hard-coded, and makes quality-of-life improvements along the way.

Following up after this PR, we may also want to make some the remaining hard-coded parameters configurable (by argparse initially, and then if / when these become too nested, considering yaml or similar):
* Parameters governing optimizer step size
* Parameters governing loss function shape

File: examples/hif2a/fit_to_multiple_rbfes.py [hunk 3]
```diff
@@ -125,20 +122,32 @@ def _save_forcefield(filename, ff_params):
 
 
 if __name__ == "__main__":
+    parser = ArgumentParser(description="Fit Forcefield parameters to hif2a")
+    parser.add_argument("--num-gpus", default=NUM_GPUS, help="Number of GPUs to run against")
+    parser.add_argument("--hosts", nargs="*", default=None, help="Hosts running GRPC worker to use for compute")
```
Comment (proteneer) [inline examples/hif2a/fit_to_multiple_rbfes.py:127]: We should probably indicate and check that --hosts and --num_gpus is an XOR relationship

File: examples/hif2a/fit_to_multiple_rbfes.py [hunk 4]
```diff
@@ -125,20 +122,32 @@ def _save_forcefield(filename, ff_params):
...
-    # set up multi-GPU client
-    client = CUDAPoolClient(max_workers=configuration.num_gpus)
+    configuration = None
+    if args.config == "intermediate": # goldilocks
+        configuration = intermediate_configuration
+    elif args.config == "test":
+        configuration = testing_configuration # a little
+    else:
+        configuration = production_configuration # a lot
+    assert configuration is not None, "No configuration provided"
```
Comment (proteneer) [inline examples/hif2a/fit_to_multiple_rbfes.py:143]: when would this ever trigger? One of the three cases for the if statement would always be true; no?

File: examples/hif2a/fit_to_multiple_rbfes.py [hunk 5]
```diff
@@ -125,20 +122,32 @@ def _save_forcefield(filename, ff_params):
...
-    # TODO: make this configurable also
-
-    # set up multi-GPU client
-    client = CUDAPoolClient(max_workers=configuration.num_gpus)
+    configuration = None
+    if args.config == "intermediate": # goldilocks
+        configuration = intermediate_configuration
+    elif args.config == "test":
+        configuration = testing_configuration # a little
+    else:
```
Comment (proteneer) [inline examples/hif2a/fit_to_multiple_rbfes.py:141]: I would probably keep the assertion below and change this to an elif for production.

**Review by proteneer — APPROVED** [review]
lgtm - I'd change the logic slightly for some of the cases to be more explicit.

File: examples/hif2a/fit_to_multiple_rbfes.py (hunk 4 above)
Comment (badisa) [reply examples/hif2a/fit_to_multiple_rbfes.py:143]: You are correct in that it would always be true as written. I added it in case the code changed and the logic was no longer valid

**Comment by proteneer** [comment]
lgtm - feel free to merge when you think its ready
